bigtable/client/cell.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_CELL_H_
+#define BIGTABLE_CLIENT_CELL_H_
+
+#include <absl/base/thread_annotations.h>
+#include <absl/strings/string_view.h>
+#include <absl/synchronization/mutex.h>
+#include <bigtable/client/version.h>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * The in-memory representation of a Bigtable cell.
+ *
+ * Bigtable stores data in rows, indexes by row keys.  Each row may contain
```

Comment:
This comment is technically correct (to my knowledge), but does not tell me anything about what I need to know about the Cell class.  Please consider an additional paragraph such as:

```
/**
 * <existing comments here>
 *
 * The Cloud Bigtable C++ Client library returns objects of this class to represent cells.
 * A Cell object can only be used while it remains valid.  During a ReadRows() iteration the
 * following actions invalidate a Cell:
 * - Reading the next row.
 * - Terminating the iteration.
 * - ....
 */
```

In other words, let people know how this class is to be used, and any caveats around it.

Reply:
Done (a bit differently).

I've switched to having the Cell hold its data. The main reason for making it hold only string_view pointers inside Row was to avoid copying the contents of ReadRowsResponse received from the wire, but it seems that the protobuf implementation already does that, by creating std::string objects for each chunk.

Accessors still return string_view, and I added a note to each about the data lifetime.

Reply:
To speed things up, let's store the consolidated value in the MVP and extend the class later if needed.